WebDriverAgentLib/Commands/FBAlertViewCommands.m

```diff
@@ -44,19 +45,51 @@ + (NSArray *)routes
 + (id<FBResponsePayload>)handleAlertAcceptCommand:(FBRouteRequest *)request
 {
   FBSession *session = request.session;
-  if (![[FBAlert alertWithApplication:session.application] acceptWithError:nil]) {
+  NSString *name = request.parameters[@"name"];
+  FBAlert *alert = [FBAlert alertWithApplication:session.application];
+  NSError *error;
+
+  if (!alert.isPresent) {
     return FBResponseWithStatus(FBCommandStatusNoAlertPresent, nil);
   }
+  if ([name length] != 0) {
+    if (![alert clickAlertButton:name error:&error]) {
+      return FBResponseWithError(error);
+    }
+  } else if (![alert acceptWithError:nil]) {
+    return FBResponseWithError(error);
+  }
   return FBResponseWithOK();
 }
 
 + (id<FBResponsePayload>)handleAlertDismissCommand:(FBRouteRequest *)request
 {
   FBSession *session = request.session;
-  if (![[FBAlert alertWithApplication:session.application] dismissWithError:nil]) {
+  NSString *name = request.parameters[@"name"];
+  FBAlert *alert = [FBAlert alertWithApplication:session.application];
+  NSError *error;
+    
+  if (!alert.isPresent) {
     return FBResponseWithStatus(FBCommandStatusNoAlertPresent, nil);
   }
+  if ([name length] != 0) {
+    if (![alert clickAlertButton:name error:&error]) {
+      return FBResponseWithError(error);
+    }
+  } else if (![alert dismissWithError:nil]) {
+    return FBResponseWithError(error);
+  }
   return FBResponseWithOK();
 }
 
++ (id<FBResponsePayload>)handleGetAlertButtonsCommand:(FBRouteRequest *)request {
+  FBSession *session = request.session;
+  FBAlert *alert = [FBAlert alertWithApplication:session.application];
+  NSArray *labels = alert.buttonLabels;
+  
+  if (!labels) {
```

Comment:
I would check of alert presence explicitly here. We would have same logic everywhere